File: Java/com.faculdadevincit.designpatterns.strategypattern/src/PagamentoComCartaoDeCredito.java
public class PagamentoComCartaoDeCredito implements IPagamentoStrategy {

    @Override
    public void pagar(double valor) {
        double acrescimo = valor * .10;
        double valorComAcrescimo = valor + acrescimo;
        System.out.println("Pagamento de R$ " + valorComAcrescimo + " realizado com sucesso!");
        System.out.println("Acréscimo de R$ " + acrescimo + " por pagamento no Cartão!");
    }

}

File: CSharp/FaculdadeVincit.DesignPatterns.StrategyPattern/PagamentoComCartaoDeCredito.cs
namespace FaculdadeVincit.DesignPatterns.StrategyPattern;

public class PagamentoComCartaoDeCredito : IPagamentoStrategy
{
    public void Pagar(double valor)
    {
        var acrescimo = valor * .10;
        var valorComAcrescimo = valor + acrescimo;
        Console.WriteLine($"Pagamento de R$ {valorComAcrescimo} realizado com Cartão!");
        System.Console.WriteLine($"Acréscimo de R$ {acrescimo} no Cartão");
    }

}
